src/robotFactory.js

```diff
@@ -1,10 +1,83 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+  }
+  goForward(value = 1) {
+    if (value > 0) {
+      this.coords.y += value;
+    }
+  }
+  goBack(value = 1) {
+    if (value > 0) {
+      this.coords.y -= value;
+    }
+  }
+  goRight(value = 1) {
+    if (value > 0) {
+      this.coords.x += value;
+    }
+  }
+  goLeft(value = 1) {
+    if (value > 0) {
+      this.coords.x -= value;
+    }
+  }
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, `
+     + `Weight: ${this.weight}`;
+  }
+}
 
-class DeliveryDrone {}
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+      z: coords.z || 0,
+    };
```

Comment:
```suggestion
    this.coords = {
      z: coords.z || 0,
    };
```
You can add only new one, as all the others were added in super